docs/reference/aggregations/pipeline/movfn-aggregation.asciidoc

```diff
@@ -0,0 +1,575 @@
+
+[source,js]
+--------------------------------------------------
+{
+   "took": 11,
+   "timed_out": false,
+   "_shards": ...,
+   "hits": ...,
+   "aggregations": {
+      "my_date_histo": {
+         "buckets": [
+             {
+                 "key_as_string": "2015/01/01 00:00:00",
+                 "key": 1420070400000,
+                 "doc_count": 3,
+                 "the_sum": {
+                    "value": 550.0
+                 }
+             },
+             {
+                 "key_as_string": "2015/02/01 00:00:00",
+                 "key": 1422748800000,
+                 "doc_count": 2,
+                 "the_sum": {
+                    "value": 60.0
+                 },
+                 "moving_fn": {
+                    "value": 550.0
+                 }
+             },
+             {
+                 "key_as_string": "2015/03/01 00:00:00",
+                 "key": 1425168000000,
+                 "doc_count": 2,
+                 "the_sum": {
+                    "value": 375.0
+                 },
+                 "moving_fn": {
+                    "value": 305.0
+                 }
+             }
+         ]
+      }
+   }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"took": 11/"took": $body.took/]
+// TESTRESPONSE[s/"_shards": \.\.\./"_shards": $body._shards/]
+// TESTRESPONSE[s/"hits": \.\.\./"hits": $body.hits/]
+
+
+==== Custom user scripting
+
+The Moving Function aggregation allows the user to specify any arbitrary script to define custom logic.  The script is invoked each time a
+new window of data is collected.  These values are provided to the script in the `values` variable.  The script should then perform some
+kind of calculation and emit a single `double` as the result.  Emitting `null` is not permitted, although `NaN` and +/- `Inf` are allowed.
+
+For example, this script will simply return the first value from the window:
+
+[source,js]
+--------------------------------------------------
+POST /_search
+{
+    "size": 0,
+    "aggs": {
+        "my_date_histo":{
+            "date_histogram":{
+                "field":"date",
+                "interval":"1M"
+            },
+            "aggs":{
+                "the_sum":{
+                    "sum":{ "field": "price" }
+                },
+                "the_movavg": {
+                    "moving_fn": {
+                        "buckets_path": "the_sum",
+                        "window": 10,
+                        "script": "return values[0];"
+                    }
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:sales]
+
+==== Pre-built Functions
+
+For convenience, a number of functions have been prebuilt and are available inside the `moving_fn` script context:
+
+- `windowMax()`
+- `windowMin()`
+- `windowSum()`
+- `simpleMovAvg()`
+- `linearMovAvg()`
+- `ewmaMovAvg()`
+- `holtMovAvg()`
+- `holtWintersMovAvg()`
```

Comment:
I wonder if it would be useful to add `windowStdDev()` to this list too as I have seen some requests for moving standard deviations too. However I imagine given that it would be used together with moving average, it would need the simple, linear, ewma, holt and holtWinters forms to make sense with moving average, so maybe its too complex to add it in this PR?